### PR TITLE
feat(metrics): Add a "custom" use case for metrics

### PR DESCRIPTION
**Pull request by jan-auer**
Implements Sentry's portion of https://github.com/getsentry/sentry/issues/55451. Custom user-defined metrics will be stored in their own namespace / use case.

See https://github.com/getsentry/relay/pull/2443 for usage of the feature flag.

### CODE REVIEW

**Review by iker-barriocanal — APPROVED** [review]

File: src/sentry/snuba/metrics/datasource.py [hunk 1]
```diff
@@ -159,6 +159,12 @@ def get_metrics(projects: Sequence[Project], use_case_id: UseCaseID) -> Sequence
             "d": "generic_metrics_distributions",
             "g": "generic_metrics_gauges",
         },
+        "custom": {
+            "c": "generic_metrics_counters",
+            "s": "generic_metrics_sets",
+            "d": "generic_metrics_distributions",
+            "g": "generic_metrics_gauges",
```
Comment (onewland) [inline src/sentry/snuba/metrics/datasource.py:166]: this absolutely doesn't need to slow down this change, but just a point of information: the metrics storage doesn't support gauges

Comment (jan-auer) [reply src/sentry/snuba/metrics/datasource.py:166]: gotcha. i copied what was also declared above; would you prefer me to remove it for now?

we might eventually want to add gauges to generic metrics, but we can definitely get going without it.

**Comment by ayirr7** [comment]
Use-case registration looks good, thanks!

File: src/sentry/snuba/metrics/datasource.py (hunk 1 above)
Comment (ayirr7) [reply src/sentry/snuba/metrics/datasource.py:166]: I think it would be fine if it remains but I will wait in case Oliver might have a different thought

**Comment by jan-auer** [comment]
Thanks, all! I'm going ahead with merging this; it's going to be easy to adapt this as we move along.

File: src/sentry/snuba/metrics/datasource.py (hunk 1 above)
Comment (onewland) [reply src/sentry/snuba/metrics/datasource.py:166]: no, I wasn't asking you to remove it (I figured it was copy-pasted) -- just wanted to make sure that you were aware that gauges aren't there today

**Pull request merged**